control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -387,14 +387,21 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		return err
 	}
 
+	_, externalTopic := isExternalTopic(broker)
 	secret, err := security.Secret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig, UseNamespaceInConfigmap: false}, r.SecretProviderFunc())
```

Comment:
maybe we can move this to the block starting on line 405, as this is only needed in case of a internal topic  :shrug:

Reply:
we still have 
```
	if err := r.removeFinalizerSecret(ctx, finalizerSecret(broker), secret); err != nil {
		return err
	}
```

on 448+

Reply:
ah ok